authoringAssistant/src/main/java/explorableviz/transparenttext/Main.java

```diff
@@ -17,53 +17,74 @@ public static void main(String... args) {
         Map<String, String> arguments = parseArguments(args);
         logger.info("Arguments passed from command line");
         logger.info(arguments.toString().replace(",", "\n"));
-        final ArrayList<QueryContext> queryContexts;
-        final LearningQueryContext learningQueryContext;
+        final ArrayList<Query> queries;
+        final LearningQuery learningQuery;
         final Optional<Integer> numQueryToExecute = arguments.containsKey("numQueryToExecute") ? Optional.of(Integer.parseInt(arguments.get("numQueryToExecute"))) : Optional.empty();
         try {
-            Settings.getInstance().loadSettings(arguments.get("settings"));
-            learningQueryContext = LearningQueryContext.importLearningCaseFromJSON(arguments.get("inContextLearningPath"), Integer.parseInt(arguments.get("numLearningCaseToGenerate")));
-            queryContexts = TestQueryContext.loadCases(arguments.get("testPath"), Integer.parseInt(arguments.get("numTestToGenerate")));
-            final int queryLimit = numQueryToExecute.orElseGet(queryContexts::size);
-            final ArrayList<String> results = execute(learningQueryContext, arguments.get("agent"), queryLimit, queryContexts);
-            if (computeAccuracy(results, queryContexts, queryLimit, Integer.parseInt(arguments.get("threshold")))) {
+            Settings.init(arguments.get("settings"));
+            learningQuery = LearningQuery.importLearningCaseFromJSON(arguments.get("inContextLearningPath"), Integer.parseInt(arguments.get("numLearningCaseToGenerate")));
+            queries = TestQuery.loadCases(arguments.get("testPath"), Integer.parseInt(arguments.get("numTestToGenerate")));
+            final int queryLimit = numQueryToExecute.orElseGet(queries::size);
+            final ArrayList<QueryResult> results = execute(learningQuery, arguments.get("agent"), queryLimit, queries);
+            float accuracy = computeAccuracy(results, queries, queryLimit);
+            writeLog(results, arguments.get("agent"), learningQuery.size());
```

Comment:
The inlining has gone a bit far here – it’s perfectly reasonable to capture each argument (once) as a variable, i.e. `final agent = arguments.get("agent")`.

Reply:
It would probably be sensible to treat arguments similarly to settings, i.e. have a class `Arguments` that does the decoding, make `parseArguments` be a static method for instantiating that class and then have one method for each `arguments.get("property")`.